core/src/main/java/org/mule/runtime/core/internal/profiling/tracing/export/OpentelemetrySpanExporter.java

```diff
@@ -59,6 +59,10 @@ public OpentelemetrySpanExporter(Tracer tracer, EventContext eventContext,
 
   @Override
   public void export(InternalSpan internalSpan) {
+    for (Map.Entry<String, String> entry : internalSpan.attributesAsMap().entrySet()) {
```

Comment:
Is this ok to be here or should we add the attributes to the OTEL span as soon as they are added to our internal span?

Reply:
In practical terms, it is the same as the export in opentelemetry happens when you set the end, but I've changed it for future developments.